worlds/civ_6/Items.py

```diff
@@ -0,0 +1,296 @@
+    "TECH_THE_WHEEL",
+    "TECH_MASONRY",
+    "TECH_ARCHERY",
+    "TECH_ENGINEERING",
+    "TECH_CONSTRUCTION",
+    "TECH_GUNPOWDER",
+    "TECH_MACHINERY",
+    "TECH_SIEGE_TACTICS",
+    "TECH_STIRRUPS",
+    "TECH_ASTRONOMY",
+    "TECH_BALLISTICS",
+    "TECH_STEAM_POWER",
+    "TECH_SANITATION",
+    "TECH_COMPUTERS",
+    "TECH_COMBUSTION",
+    "TECH_TELECOMMUNICATIONS",
+    "TECH_ROBOTICS",
+    "CIVIC_FEUDALISM",
+    "CIVIC_GUILDS",
+    "CIVIC_THE_ENLIGHTENMENT",
+    "CIVIC_MERCANTILISM",
+    "CIVIC_CONSERVATION",
+    "CIVIC_CIVIL_SERVICE",
+    "CIVIC_GLOBALIZATION",
+    "CIVIC_COLD_WAR",
+    "CIVIC_URBANIZATION",
+    "CIVIC_NATIONALISM",
+    "CIVIC_MOBILIZATION",
+    "PROGRESSIVE_NEIGHBORHOOD",
+    "PROGRESSIVE_PRESERVE"
+]
+
+
+class FillerItemRarity(Enum):
+    COMMON = "COMMON"
+    UNCOMMON = "UNCOMMON"
+    RARE = "RARE"
+
+
+FILLER_DISTRIBUTION: Dict[FillerItemRarity, float] = {
+    FillerItemRarity.RARE: 0.025,
+    FillerItemRarity.UNCOMMON: .2,
+    FillerItemRarity.COMMON: 0.775,
+}
+
+
+class FillerItemData:
+    name: str
+    type: str
+    rarity: FillerItemRarity
+    civ_name: str
+
+    def __init__(self, data: Dict[str, str]):
+        self.name = data["Name"]
+        self.rarity = FillerItemRarity(data["Rarity"])
+        self.civ_name = data["Type"]
+
+
+def get_filler_item_data() -> Dict[str, FillerItemData]:
+    """
+    Returns a dictionary of filler items with their data
+    """
+    goody_huts: List[Dict[str, str]] = get_goody_hut_rewards_data()
+    # Create a FillerItemData object for each item
+    cached_filler_items = {item["Name"]: FillerItemData(item) for item in goody_huts}
+
+    return cached_filler_items
+
+
+class CivVIItemData:
+    civ_vi_id: int
+    classification: ItemClassification
+    name: str
+    code: int
+    cost: int
+    item_type: CivVICheckType
+    progression_name: Optional[str]
+    civ_name: Optional[str]
+
+    def __init__(self, name, civ_vi_id: int, cost: int, item_type: CivVICheckType, id_offset: int, classification: ItemClassification, progression_name: Optional[str], civ_name: Optional[str] = None):
+        self.classification = classification
+        self.civ_vi_id = civ_vi_id
+        self.name = name
+        self.code = civ_vi_id + CIV_VI_AP_ITEM_ID_BASE + id_offset
+        self.cost = cost
+        self.item_type = item_type
+        self.progression_name = progression_name
+        self.civ_name = civ_name
+
+
+class CivVIItem(Item):
+    game: str = "Civilization VI"
+    civ_vi_id: int
+    item_type: CivVICheckType
+
+    def __init__(self, item: CivVIItemData, player: int, classification: ItemClassification = None):
+        super().__init__(item.name, classification or item.classification, item.code, player)
+        self.civ_vi_id = item.civ_vi_id
+        self.item_type = item.item_type
+
+
+def format_item_name(name: str) -> str:
+    name_parts = name.split("_")
+    return " ".join([part.capitalize() for part in name_parts])

```

Comment:
```py
def format_item_name(name: str) -> str:
    return name.replace("_", " ").title()
```
I think the way you're doing it is fine, though. Was mostly trying to figure out a different way to do it lol

Though also, can you explain why you have the names as "TECH_THE_WHEEL" and then convert them to "Tech The Wheel" instead of just having them as "Tech The Wheel" in the first place? Mostly curious

Reply:
A solid curiosity lol Civ and the mod treat these names as IDs so it initially made sense to just keep them as the ugly format for the data side of things. I'd probably approach it slightly differently if I was to start again 🤔

Reply:
That's fair!